gpytorch/likelihoods/multitask_gaussian_likelihood.py

```diff
@@ -31,31 +31,34 @@ def __init__(
             `rank` > 0, or a prior over the log of just the diagonal elements, if `rank` == 0.
 
         """
-        super(MultitaskGaussianLikelihood, self).__init__(batch_size=batch_size, log_noise_prior=log_noise_prior)
+        super(MultitaskGaussianLikelihood, self).__init__(
+            batch_size=batch_size, log_noise_prior=log_noise_prior, param_transform=param_transform
+        )
 
         if rank == 0:
             self.register_parameter(
                 name="log_task_noises",
                 parameter=torch.nn.Parameter(torch.zeros(batch_size, num_tasks)),
                 prior=task_prior,
+                transform=param_transform,
             )
         else:
             self.register_parameter(
                 name="task_noise_covar_factor", parameter=torch.nn.Parameter(torch.randn(batch_size, num_tasks, rank))
             )
             if task_prior is not None:
-                self.register_derived_prior(
-                    name="MultitaskErrorCovariancePrior",
-                    prior=task_prior,
-                    parameter_names=("task_noise_covar_factor", "log_noise"),
+                self.register_prior(
+                    "MultitaskErrorCovariancePrior",
+                    task_prior,
+                    "task_noise_covar_factor",
+                    "log_noise",
```

Comment:
With the new suggested interface, I think you could do `register_prior("MultitaskErrorCovariancePrior", task_prior, "_eval_covar_matrix")`.

`_eval_covar_matrix` would probably be re-written so that it takes no arguments, which should be fine since `task_noise_covar_factor` and `log_noise` are parameters that belong to the likelihood anyways.